### DISCUSSION TITLE
Static progress bar

### DISCUSSION THREAD

**Discussion by adfaure** [Textualize/rich#2906] [Q&A] [answered]
Hello,
I would like to render progress bars that are static, I don't need to update it during the application.
However the `Progress` class seems centered around the idea of updates.

I found the class `ProgressBar` that seems to be what I need (it even works in a table which is what I need). But, I am concerned that the user documentation (which is great by the way) doesn't mention this class at all.

I am on the right track, or should I use the `Progress` class ?
Sorry if my question sounds silly.

Thank you !

**Comment by willmcgugan** [comment, marked as answer]
You could construct a Progress, and print it. That will give you the same output, but it won't update.

Or you could use the `Progress` renderable for a simple bar.

**Reply by adfaure** [reply]
I tried this approach, however I did not find how I could give it a current progress with pre defined completed status (such as the completed arg ins ProgressBar `__init__`:         `pb = ProgressBar(total = number_of_resources, completed=number_of_busy_resources)`).

> Or you could use the Progress renderable for a simple bar.
I am sorry, but I don't understand this option.

**Reply by adfaure** [reply]
I think I found a way to do what you said with:

```python
progress = Progress(text_column, bar_column, expand=True)
task = progress.add_task(f"{number_of_busy_resources}/{number_of_resources}", total=)
progress.advance(0, number_of_busy_resources)

# Add the progress in my table
table.add_row(net, progress)
```

I think I'll go with that, thanks !
